heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java

```diff
@@ -124,27 +122,38 @@ public Response uploadFile(
  */
   @GET
   @Path("/download/{file}")
-  public Response downloadPdfFile(final @PathParam("file") String file) {
+  public Response downloadFile(final @PathParam("file") String file) {
     Config config = createConfig();
     String uploadDir = config.getStringValue(FILE_SYSTEM_DIRECTORY);
     String filePath = uploadDir + "/" + file;
     if (!new File(filePath).exists()) {
       LOG.debug("Download request file " + file + " doesn't exist at " + uploadDir);
       return Response.status(Response.Status.NOT_FOUND).build();
     }
-    StreamingOutput fileStream = new StreamingOutput() {
-      @Override
-      public void write(java.io.OutputStream output) throws IOException {
-          java.nio.file.Path path = Paths.get(filePath);
-          byte[] data = Files.readAllBytes(path);
-          output.write(data);
-          output.flush();
-      }
-    };
-    return Response
-        .ok(fileStream, MediaType.APPLICATION_OCTET_STREAM)
-        .header("content-disposition", "attachment; filename = " + file)
-        .build();
+
+    String mimeType = new MimetypesFileTypeMap().getContentType(file);
+    Response.ResponseBuilder rb = Response.ok(file, mimeType);
+    rb.header("content-disposition", "attachment; filename = "
+        + file);
+    return rb.build();
+  }
+
+  /**
+   * Endpoint for downloading Heron Core
+   */
+  @GET
+  @Path("/downloadcore")
```

Comment:
should we have something like /download/core rather /downloadcore

Reply:
I rather it not be under the same endpoint as the topology endpoint in case we have classes

Reply:
I can make that change